kpi/views/v2/asset_snapshot.py

```diff
@@ -126,6 +150,40 @@ def preview(self, request, *args, **kwargs):
             response_data = copy.copy(snapshot.details)
             return Response(response_data, template_name='preview_error.html')
 
+    @action(
+        detail=True,
+        trailing_slash='',
+        permission_classes=[EditSubmissionPermission],
+        methods=['HEAD', 'POST'],
+        authentication_classes=[DigestAuthentication],
+    )
+    def submission(self, request, *args, **kwargs):
+        if request.method == 'HEAD':
+            # ToDo figure out what's the valid response
+            return Response({})
```

Comment:
Empty response body is okay. The response code should be 204. Some headers are supposed to be returned as well: `X-OpenRosa-Version`, `X-OpenRosa-Accept-Content-Length`. See https://docs.getodk.org/openrosa-form-submission/#extended-transmission-considerations